zlib/grabber.go

```diff
@@ -238,6 +238,8 @@ func makeGrabber(config *Config) func(*Conn) error {
 			config.ErrorLog.Errorf("Conversation error with remote host %s: %s",
 				c.RemoteAddr().String(), err.Error())
 		}
+
+		c.Close()
```

Comment:
We should also figure out how to make this work with only IP as long as the
redirect IP matches the original one pasted in (no domain was ever
specified)
On Tue, Oct 20, 2015 at 5:17 AM Zane Ma <notifications@github.com> wrote:

> In zlib/grabber.go
> <https://github.com/zmap/zgrab/pull/69#discussion_r42472297>:
>
> > @@ -238,6 +238,8 @@ func makeGrabber(config *Config) func(*Conn) error {
> >  			config.ErrorLog.Errorf("Conversation error with remote host %s: %s",
> >  				c.RemoteAddr().String(), err.Error())
> >  		}
> > +
> > +		c.Close()
>
> From what I could tell, we were never explicitly closing any of our HTTP
> connections. This seems important because we are doing HTTP 1.1 in all our
> initial requests, which defaults to keep-alive connections.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zmap/zgrab/pull/69/files#r42472297>.
>